community/sinit/files/etc/rc.d/sinit.pre.shutdown

```diff
@@ -0,0 +1,10 @@
+if [ "$SINIT_ENABLE_SHUTDOWN_HOOKS" = 1 ]; then
+    case "$1" in
+        reboot|poweroff)
+        ;;
+        *)
+            echo "Invalid action '$1' for rc.shutdown" 1>&2
+            exit 1
+            ;;
+    esac
+fi
```

Comment:
What is the point of this file?

Reply:
It's from morpheus rc scripts. It did actually save me while I was testing. If your $1 is neither reboot nor poweroff all your services will be stopped but the computer will stay turned on, if this check doesn't happen. Could be removed though.

Post shutdown hook could be changed to

```sh
case "$1" in
        reboot) halt -r ;;
        *) halt -p ;;
esac
```

so that it doesn't require the check on pre shutdown

Reply:
It has saved me a couple times, as well, while tinkering.

Reply:
> It's from morpheus rc scripts. It did actually save me while I was testing. If your $1 is neither reboot nor poweroff all your services will be stopped but the computer will stay turned on, if this check doesn't happen. Could be removed though.
> 
> Post shutdown hook could be changed to
> 
> ```shell
> case "$1" in
>         reboot) halt -r ;;
>         *) halt -p ;;
> esac
> ```
> 
> so that it doesn't require the check on pre shutdown

Regarding your change on the post shutdown hook, it needs to be guarded 

`if [ "$SINIT_ENABLE_SHUTDOWN_HOOKS" = 1 ]; then ...`

because a user could install sinit, without intentions of configuring it immediately, but the post shutdown hook would be running and conflicting with the current busybox configuration.

Reply:
Yeah, you are right, I just wanted to keep the comment short, I did not mean to remove the check.